airbyte-commons/src/main/java/io/airbyte/commons/features/EnvVariableFeatureFlags.java

```diff
@@ -26,4 +27,20 @@ public boolean forceSecretMigration() {
     return Boolean.parseBoolean(System.getenv("FORCE_MIGRATE_SECRET_STORE"));
   }
 
+  @Override
+  public boolean useStreamCapableState() {
+    return getEnvOrDefault("USE_STREAM_CAPABLE_STATE", false, Boolean::parseBoolean);
+  }
+
+  // TODO: refactor in order to use the same method than the ones in EnvConfigs.java
+  public <T> T getEnvOrDefault(final String key, final T defaultValue, final Function<String, T> parser) {
```

Comment:
Why not just put this environment variable in `EnvConfigs`?

Reply:
I felt that it was relevant to have them as a flag since it is not really a config but a switch on a functionality we want.